src/data-workspace/data-entry-cell/value-types.js

```diff
@@ -0,0 +1,126 @@
+import {
+    email,
+    number,
+    integer,
+    internationalPhoneNumber,
+    url,
+} from '@dhis2/ui-forms'
+import { FileResourceInput, ImageInput } from './file-inputs.js'
+import {
+    BasicInput,
+    TrueOnlyCheckbox,
+    BooleanRadios,
+    OptionSet,
+    withAdditionalProps,
+    LongText,
+} from './inputs.js'
+import {
+    text,
+    letter,
+    integerPositive,
+    integerZeroOrPositive,
+    integerNegative,
+    percentage,
+    unitInterval,
+    // date,
+    time,
+    // dateTime,
+} from './validators.js'
+
+/** Use an OptionSet if relevant; else use input type below */
+export function getInputByDataElement(de) {
+    if (de.optionSetValue) {
+        return OptionSet
+    } else {
+        return VALUE_TYPES[de.valueType].Input
+    }
+}
```

Comment:
The pattern I can see here is a pattern I have seen quite often in dhis2 now and I think we should try to avoid it as it's historically always became a mess, I'd also say that it's an anti-pattern to the react way (at least that's my experience.

The approach here is that there's a config (`VALUE_TYPES`) and the data entry cell can dynamically get what it needs. I'd call that the config based approach.

IMO the "correct" react-style approach would be to have components like `TextInput`, `LongTextInput`, etc.
The `getInputByDataElement` function would turn into a component that would have to be used like this: `<DataEntryCellInput isOptionSetValue={!!de.optionSetValue} valueType={de.valueType} />`, which renders one of the components below.

It'd be more boilerplate code, but it's the most maintainable style imo, as you can easily handle exceptions that will have to be added eventually at an unknown time (might be during QA testing, might be due to backend changes).
It's also easier to reason about the code that way, despite it having more LoCs.
Especially as the contents of the `value-types.js` are being used by the `data-entry-cell.js` module, I think a configuration based approach just leads to code that's harder to understand in the long run